app/src/common/shared/com/igalia/wolvic/ui/widgets/menus/ContextMenuWidget.java

```diff
@@ -101,7 +101,7 @@ public void setContextElement(WSession.ContentDelegate.ContextElement aContextEl
             // Open link in a new tab
             mItems.add(new MenuWidget.MenuItem(getContext().getString(R.string.context_menu_open_link_new_tab_1), 0, () -> {
                 if (!StringUtils.isEmpty(aContextElement.linkUri)) {
-                    widgetManager.openNewTab(aContextElement.linkUri);
+                    widgetManager.openNewTabForeground(aContextElement.linkUri);
```

Comment:
Why this?

Reply:
In the current architecture `SessionStore` simply passes events from each individual session to the tabs widget.

When a tab is opened in the background, its session does not call the `onSessionAdded()` event and therefore the tabs widget is not updated.

I have tried forcing a call to `SessionStore.onSessionAdded()` but then I get a crash in Gecko: "IllegalArgumentException: Tab with same ID already exists".

I am not sure but the problem might be that the background session is in an incomplete state until it is first shown.